src/rhsm/connection.py

```diff
@@ -1345,20 +1345,56 @@ def regenIdCertificate(self, consumerId):
         return self.conn.request_post(method)
 
     def regenEntitlementCertificates(self, consumer_id, lazy_regen=True):
+        """
+        Regenerates all entitlements for the given consumer
+        """
+
         method = "/consumers/%s/certificates" % self.sanitize(consumer_id)
 
         if lazy_regen:
             method += "?lazy_regen=true"
 
-        return self.conn.request_put(method)
+        result = False
+
+        try:
+            self.conn.request_put(method)
+            result = True
+        except RemoteServerException as e:
```

Comment:
For this to work, if the resource is truly not available, we should probably be catching BadStatusLine.